### PR TITLE
Use the fully qualified registry name in state

### PR DESCRIPTION
**Pull request by SludgeGirl**
When using a registry like `registry.opensuse.org` it would trim off `registry.` for the ui, then later on when it came to actually filtering the results it would also only select `images[opensuse.org]` which would never exist

### CODE REVIEW

**Comment by jelly** [comment]
> When using a registry like `registry.opensuse.org` it would trim off `registry.` for the ui, then later on when it came to actually filtering the results it would also only select `images[opensuse.org]` which would never exist

Does this mean you configure "registry.opensuse.org" in unqualified-search-registries?

**Comment by SludgeGirl** [comment]
> We will also want to add an integration test here so it doesn't regress

Gotcha I'll get one added now

**Comment by SludgeGirl** [comment]
@jelly it's a little clunky so let me know if there's a better way I can handle it

File: test/check-application [hunk 1]
```diff
@@ -1725,6 +1735,11 @@ PodName={name}
         b.wait_not_present(f'.pf-v6-c-menu__item:contains("{IMG_REGISTRY_LATEST}")')
         b.wait_visible(f'.pf-v6-c-menu__item:contains("{IMG_ALPINE_LATEST}")')
 
+        # Test filter with a subdomain.
+        b.set_input_text("#create-image-image input", "my-busybox", blur=False)
+        b.click('.pf-v6-c-toggle-group__text:contains("local.localhost")')
+        b.wait_text(".pf-v6-c-menu__item:not([disabled])", "subdomain.local.localhost:7000/my-busybox")
```
Comment (jelly) [inline test/check-application:1741]: I think we can use the existing registries here for filtering?

image


vs.

image

Comment (jelly) [reply test/check-application:1741]: I manually tested the test without your fixes and it indeed does not filter on `localhost:6000` or `5000` so that should be good enough.

**Review by jelly — CHANGES_REQUESTED** [review]
Thanks, some small test fixes and then this looks good :+1:

File: test/check-application (hunk 1 above)
Comment (SludgeGirl) [reply test/check-application:1741]: I've checked myself against main and it does filter correctly on both of those

image

The only changes we make to the domain which causes this problem is:

https://github.com/cockpit-project/cockpit-podman/blob/28483f585b06634cb186a4cb453c6e0f409a729e/src/ImageRunModal.jsx#L625-L632

So if we just ran the checks against `localhost:6000` or `:5000` we wouldn't really be testing for a regression, it needs to have a subdomain

Comment (jelly) [reply test/check-application:1741]: I would recheck but adding new extra test coverage doesn't hurt and would be nice! So please keep that in and drop the empty `testBasicFilters` test

https://cockpit-logs.us-east-1.linodeobjects.com/pull-2092-79bbb487-20250415-085646-fedora-41/log.html

**Review by jelly — DISMISSED** [review]

**Comment by jelly** [comment]
And bah ubuntu-2204 has a broken podman release, search returns:

```
Index:  "local.localhost:7000"
Name:  "subdomain.local.localhost:7000/my-busybox"
```

So it just drops off the registry name for no reason. There is no way to guess what is right and I don't want to hold up merging this.

**Review by jelly — APPROVED** [review]
